tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -0,0 +1,410 @@
+	// Ensure that the Destination/Server cluster is using file based tls configs
+	err = destCluster.Exists("{.cluster.transportSocket.typedConfig.commonTlsContext.tlsCertificateSdsSecretConfigs[?(@.name == 'file-cert:/client-certs/cert.pem~/client-certs/key.pem')]}}").Check()
+	if err != nil {
+		t.Fatalf("Destination cluster on the Client side is expected to use FileBased SDS configs. %v", err)
+	}
+
+
+	// Verify server side configs
+	serverWorkloads, _ := server.Workloads()
+	if serverWorkloads[0].Sidecar() == nil {
+		t.Fatalf("Sidcecar is expected, but was not found")
+	}
+
+	configDump, err = serverWorkloads[0].Sidecar().Config()
+	if err != nil {
+		t.Fatalf("Unable to retrieve config_dump from the server. %v", err)
+	}
+
+	validator = structpath.ForProto(configDump)
+	virtualInboundListener := validator.Select("{.configs[*].dynamicListeners[?(@.name == 'virtualInbound')]}")
+	err = virtualInboundListener.Select("{.activeState.listener.filterChains[*].transportSocket.typedConfig.commonTlsContext.tlsCertificateSdsSecretConfigs[?(@.name == 'file-cert:/server-certs/cert.pem~/server-certs/key.pem')]}").Check()
+	if err != nil {
+		t.Fatalf("Server is expected to use FileBased SDS configs with DownstreamTlsContext. %v", err)
+	}
+}
+
+func verifyStats(t *testing.T, client echo.Instance, server echo.Instance) {
+	// Verify client side stats
+	clientWorkloads, _ := client.Workloads()
+	if clientWorkloads[0].Sidecar() == nil {
+		t.Fatalf("Sidcecar is expected, but was not found")
+	}
+
+	stats := clientWorkloads[0].Sidecar().StatsOrFail(t)
+
+	clusterName := clusterName(server, echo.Port{ServicePort: 8443})
+	var metric *dto.Metric
+
+	clientTestCases := []struct {
+		metricName string
+		expectedValue float64
+	}{
+		{
+			metricName: "envoy_cluster_ssl_handshake",
+			expectedValue: 1.0,
+		},
+		{
+			metricName: "envoy_cluster_upstream_rq_200",
+			expectedValue: 1.0,
+		},
+	}
+
+	for _, tc := range clientTestCases {
+		metric = getMetricForCluster(stats[tc.metricName].Metric, clusterName)
+		if *metric.Counter.Value < tc.expectedValue {
+			t.Fatalf("%s expected, but not found. Expected - %v, Got - %v", tc.metricName,  tc.expectedValue, metric.Counter.Value)
+		}
+	}
+
+
+	// Verify server side stats
+	serverWorkloads, _ := server.Workloads()
+	if serverWorkloads[0].Sidecar() == nil {
+		t.Fatalf("Sidcecar is expected, but was not found")
+	}
+
+	stats = serverWorkloads[0].Sidecar().StatsOrFail(t)
+
+	serverTestCases := []struct {
+		metricName string
+		expectedValue float64
+	}{
+		{
+			metricName: "envoy_listener_ssl_handshake",
+			expectedValue: 1.0,
+		},
+		{
+			metricName: "envoy_listener_ssl_versions_TLSv1_2",
+			expectedValue: 1.0,
+		},
+		{
+			metricName: "envoy_listener_http_inbound_0_0_0_0_8443_downstream_rq_completed",
+			expectedValue: 1.0,
+		},
+	}
+
+	for _, tc := range serverTestCases {
+		metric = getMetricForListener(stats[tc.metricName].Metric, "0.0.0.0_15006") // Check metrics on VirtualInbound listener
```

Comment:
This seems fairly complex. Do we get any value checking metrics here, that the above testing with setting STRICT and checking for the client cert header do not give us?

Reply:
Thinking about this more - you right. STRICT + proper xfcc validation will be enough.
I just need to provide properly tailored certs.
Right now this test is reusing DNS&Pilot certs in different combinations. Let me update this.
Thank you!